src/module.rs

```diff
@@ -105,7 +110,8 @@ impl<'a> Module<'a> {
     pub fn is_empty(&self) -> bool {
         self.segments
             .iter()
-            .all(|segment| segment.value.trim().is_empty())
+            // no trim: if we add spaces/linebreaks it's not "empty" as we change the final output
+            .all(|segment| segment.value.is_empty())
```

Comment:
```suggestion
           .all(|segment| segment.value.chars().any(|c| c != ' '))
```
Since this is mostly about the line_break I think it makes sense to still treat space only segments as empty.

Reply:
Opposing argument is that even a single space shows up on the prompt and is clearly visible and therefore the corresponding module not "empty"

Reply:
Yes that makes sense.